html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Drupal\hpc_api\Query;
+
+/**
+ * Get plan information.
+ */
+class EndpointPlanQuery extends EndpointQuery {
+
+  /**
+   * Get plan.
+   */
+  public function getPlan($plan_id) {
+    $this->setEndpoint('public/plan/id/' . $plan_id);
```

Comment:
At least for Insights this won't be sufficient, as it also provides logged in users with non-public data. Or the method should be renamed to reflect that it's fetching public data.